src/main/java/core/basesyntax/utils/generate/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.utils.generate.impl;
+
+import core.basesyntax.storage.impl.StorageImpl;
+import core.basesyntax.utils.generate.ReportGenerator;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private final ReportGenerator generator;
```

Comment:
It seems like you are trying to use the 'Decorator' or 'Composite' pattern here, but this is not suitable for the `ReportGenerator` since it should be independent. Please remove the field `generator` and implement the report generation logic directly in the `getReport` method.